From 318922f2b888ce0ff5724d872c196e7d42d0710b Mon Sep 17 00:00:00 2001
From: Cristian Canedo <ccanedo@gmx.com>
Date: Fri, 4 Feb 2022 22:59:33 -0500
Subject: [PATCH] Add bookmark functionality

---
 config.def.h | 31 ++++++++++++++++++-------------
 1 file changed, 18 insertions(+), 13 deletions(-)

diff --git a/config.def.h b/config.def.h
index 1355ba3..574d93a 100644
--- a/config.def.h
+++ b/config.def.h
@@ -65,16 +65,17 @@ static WebKitFindOptions findopts = WEBKIT_FIND_OPTIONS_CASE_INSENSITIVE |
 #define PROMPT_FIND "Find:"
 
 /* SETPROP(readprop, setprop, prompt)*/
-#define SETPROP(r, s, p) { \
-        .v = (const char *[]){ "/bin/sh", "-c", \
-             "prop=\"$(printf '%b' \"$(xprop -id $1 "r" " \
-             "| sed -e 's/^"r"(UTF8_STRING) = \"\\(.*\\)\"/\\1/' " \
-             "      -e 's/\\\\\\(.\\)/\\1/g')\" " \
-             "| dmenu -p '"p"' -w $1)\" " \
-             "&& xprop -id $1 -f "s" 8u -set "s" \"$prop\"", \
-             "surf-setprop", winid, NULL \
-        } \
-}
+/* #define SETPROP(r, s, p) { \ */
+/*         .v = (const char *[]){ "/bin/sh", "-c", \ */
+/*              "prop=\"$(printf '%b' \"$(xprop -id $1 "r" " \ */
+/*              "| sed -e 's/^"r"(UTF8_STRING) = \"\\(.*\\)\"/\\1/' " \ */
+/*              "      -e 's/\\\\\\(.\\)/\\1/g')\" " \ */
+/*              "| dmenu -p '"p"' -w $1)\" " \ */
+/*              "&& xprop -id $1 -f "s" 8u -set "s" \"$prop\"", \ */
+/*              "surf-setprop", winid, NULL \ */
+/*         } \ */
+/* } */
+#define SETPROP(p) { .v = (char *[]){ "/bin/sh", "-c", "surfbmark.sh $0 $1", winid, p, NULL } }
 
 /* DOWNLOAD(URI, referer) */
 #define DOWNLOAD(u, r) { \
@@ -130,9 +131,13 @@ static SiteSpecific certs[] = {
  */
 static Key keys[] = {
 	/* modifier              keyval          function    arg */
-	{ MODKEY,                GDK_KEY_g,      spawn,      SETPROP("_SURF_URI", "_SURF_GO", PROMPT_GO) },
-	{ MODKEY,                GDK_KEY_f,      spawn,      SETPROP("_SURF_FIND", "_SURF_FIND", PROMPT_FIND) },
-	{ MODKEY,                GDK_KEY_slash,  spawn,      SETPROP("_SURF_FIND", "_SURF_FIND", PROMPT_FIND) },
+	/* { MODKEY,                GDK_KEY_g,      spawn,      SETPROP("_SURF_URI", "_SURF_GO", PROMPT_GO) }, */
+	/* { MODKEY,                GDK_KEY_f,      spawn,      SETPROP("_SURF_FIND", "_SURF_FIND", PROMPT_FIND) }, */
+	/* { MODKEY,                GDK_KEY_slash,  spawn,      SETPROP("_SURF_FIND", "_SURF_FIND", PROMPT_FIND) }, */
+	{ MODKEY,              	 GDK_g,      	 spawn,      SETPROP("_SURF_URI") },
+	{ MODKEY|GDK_SHIFT_MASK, GDK_g,      	 spawn,      SETPROP("_SURF_URI_RAW") },
+	{ MODKEY,                GDK_f,      	 spawn,      SETPROP("_SURF_FIND") },
+	{ MODKEY,                GDK_b,      	 spawn,      SETPROP("_SURF_BMARK") },
 
 	{ 0,                     GDK_KEY_Escape, stop,       { 0 } },
 	{ MODKEY,                GDK_KEY_c,      stop,       { 0 } },
-- 
2.35.1

